### PR TITLE
Fix floating point handling in Java

### PR DESCRIPTION
**Pull request by drinckes**
Updates Java library to pass tests. Addresses issues #345 and #307.

Removes BigDecimal by using integer (longs) calculation for encoding and decoding.

(This change is built on #346.)

### CODE REVIEW

**Comment by drinckes** [comment]
This implementation is significantly faster than the old one:
* encoding is almost 30 times faster (0.253usec vs 6.983usec)
* decoding is slightly faster (2.814usec vs 3.293usec).

(Both times from the TravisCI tests.)

### Previous:
```
.Encode 1000000 loops in 6982908 usecs, 6.983 usec per call
.Decode 1000000 loops in 3293220 usecs, 3.293 usec per call
```

### New:
```
.Encode 1000000 loops in 253486 usecs, 0.253 usec per call
.Decode 1000000 loops in 2814156 usecs, 2.814 usec per call
```

**Comment by drinckes** [comment]
Hi @zongweil - this change builds on the change in #346 - if you could review that one first, I'll deal with the merge fun.

File: java/src/main/java/com/google/openlocationcode/OpenLocationCode.java [hunk 1]
```diff
@@ -289,47 +302,40 @@ public CodeArea decode() {
...
-        int row = (int) (digitVal / GRID_COLUMNS.intValue());
-        int col = digitVal % GRID_COLUMNS.intValue();
-        latPrecision = latPrecision.divide(GRID_ROWS);
-        lngPrecision = lngPrecision.divide(GRID_COLUMNS);
-        southLatitude = southLatitude.add(latPrecision.multiply(new BigDecimal(row)));
-        westLongitude = westLongitude.add(lngPrecision.multiply(new BigDecimal(col)));
-        digit += 1;
-      }
-    }
+    // Initialise the values. We work them out as integers and convert them to floats at the end.
```
Comment (zongweil) [inline java/src/main/java/com/google/openlocationcode/OpenLocationCode.java]: "convert them to doubles"

Comment (drinckes) [reply java/src/main/java/com/google/openlocationcode/OpenLocationCode.java]: Done!

**Pull request merged**